lib/temporal/workflow/dispatcher.rb

```diff
@@ -39,9 +57,10 @@ def dispatch(target, event_name, args = nil)
 
       def handlers_for(target, event_name)
         handlers[target]
-          .concat(handlers[TARGET_WILDCARD])
-          .select { |event_struct| match?(event_struct, event_name) }
-          .map(&:handler)
+          .merge(handlers[TARGET_WILDCARD]) { raise 'Cannot resolve duplicate dispatcher handler IDs' }
```

Comment:
Recommend creating a real Error here instead of raising StandardError. This doesn't play well with Sentry and other error management tools when errors are non-unique.

e.g.
```
DuplicateHandlerIdError = Class.new(StandardError)
...
.merge(handlers[TARGET_WILDCARD]) { raise DuplicateHandlerIdError.new("Cannot resolve duplicate dispatcher handler IDs") }
```